pkg/provider/beyondtrust/provider.go

```diff
@@ -0,0 +1,352 @@
+		return ref.Value, nil
+	}
+
+	if err := validateSecretRef(ref); err != nil {
+		return "", err
+	}
+
+	namespace := defaultNamespace
+	if ref.SecretRef.Namespace != nil {
+		namespace = *ref.SecretRef.Namespace
+	}
+
+	ESOLogger.Info("using k8s secret", "name:", ref.SecretRef.Name, "namespace:", namespace)
+	objKey := client.ObjectKey{Namespace: namespace, Name: ref.SecretRef.Name}
+	secret := v1.Secret{}
+	err := kube.Get(ctx, objKey, &secret)
+	if err != nil {
+		return "", err
+	}
+
+	value, ok := secret.Data[ref.SecretRef.Key]
+	if !ok {
+		return "", fmt.Errorf(errNoSuchKeyFmt, ref.SecretRef.Key)
+	}
+
+	return string(value), nil
+}
+
+func validateSecretRef(ref *esv1beta1.BeyondTrustProviderSecretRef) error {
+	if ref.SecretRef != nil {
+		if ref.Value != "" {
+			return errSecretRefAndValueConflict
+		}
+		if ref.SecretRef.Name == "" {
+			return errMissingSecretName
+		}
+		if ref.SecretRef.Key == "" {
+			return errMissingSecretKey
+		}
+	} else if ref.Value == "" {
+		return errSecretRefAndValueMissing
+	}
+
+	return nil
+}
+
+func (p *Provider) GetAllSecrets(_ context.Context, _ esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	return nil, fmt.Errorf("GetAllSecrets not implemented")
+}
+
+// GetSecret reads the secret from the Password Safe server and returns it. The controller uses the value here to
+// create the Kubernetes secret.
+func (p *Provider) GetSecret(_ context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	managedAccountType := !strings.EqualFold(p.retrievaltype, "SECRET")
+
+	retrievalPaths := utils.ValidatePaths([]string{ref.Key}, managedAccountType, p.separator, &p.log)
+
+	if len(retrievalPaths) != 1 {
+		return nil, fmt.Errorf(errInvalidRetrievalPath)
+	}
+
+	retrievalPath := retrievalPaths[0]
+
+	_, err := p.authenticate.GetPasswordSafeAuthentication()
+	if err != nil {
+		return nil, fmt.Errorf("error: %w", err)
+	}
+
+	secret := keyValue{
+		Key:   "secret",
+		Value: "",
+	}
+
+	if managedAccountType {
+		var returnSecret string
+		ESOLogger.Info("retrieve managed account value", "retrievalPath:", retrievalPath)
+		manageAccountObj, _ := managed_account.NewManagedAccountObj(p.authenticate, &p.log)
+		returnSecret, err := manageAccountObj.GetSecret(retrievalPath, p.separator)
+		if err != nil {
+			err = p.authenticate.SignOut()
+			if err != nil {
+				return nil, fmt.Errorf("error: %w", err)
+			}
+			return nil, fmt.Errorf("error: %w", err)
+		}
+		secret.Value = returnSecret
+	} else {
+		var returnSecret string
+		ESOLogger.Info("retrieve secrets safe value", "retrievalPath:", retrievalPath)
+		secretObj, _ := secrets.NewSecretObj(p.authenticate, &p.log, maxFileSecretSizeBytes)
+		returnSecret, err := secretObj.GetSecret(retrievalPath, p.separator)
+		if err != nil {
+			err = p.authenticate.SignOut()
+			if err != nil {
+				return nil, fmt.Errorf("error: %w", err)
+			}
+			return nil, fmt.Errorf("error: %w", err)
+		}
+		secret.Value = returnSecret
+	}
```

Comment:
are these two going to ever be the only managedAccountTypes ? If not, I'd say replace this with a switch/case.

Reply:
Yes, we have just two options MANAGED ACCOUNT and SECRET, if it is not managed account type so it is secret type, just one of those two options.